src/main/java/uk/ac/cam/cl/dtg/segue/api/managers/UserAccountManager.java

```diff
@@ -562,21 +562,9 @@ public Response updateUserObject(final HttpServletRequest request, final HttpSer
             RegisteredUserDTO existingUserFromDb = this.getUserDTOById(userObjectFromClient
                     .getId());
 
-            if (Role.EVENT_MANAGER.equals(currentlyLoggedInUser.getRole())) {
-                if (Role.ADMIN.equals(existingUserFromDb.getRole())
-                        || Role.ADMIN.equals(userObjectFromClient.getRole())) {
-                    return new SegueErrorResponse(Response.Status.FORBIDDEN,
-                            "You cannot modify admin roles.").toResponse();
-                }
-            }
-
-            // check that the user is allowed to change the role of another user
-            // if that is what they are doing.
-            if (!checkUserRole(currentlyLoggedInUser, Arrays.asList(Role.ADMIN, Role.EVENT_MANAGER))
-                    && userObjectFromClient.getRole() != null
-                    && !userObjectFromClient.getRole().equals(existingUserFromDb.getRole())) {
+            if (null == userObjectFromClient.getRole() || !existingUserFromDb.getRole().equals(userObjectFromClient.getRole())) {
```

Comment:
This worried me, until I read it more carefully. Could we have a comment like "You cannot modify role using this endpoint" to make it more clear what this is doing?

Reply:
I thought that the message included in the error response immediately afterwards was enough to indicate what the intention of this `if` statement was. Happy to include a comment though 👍